### PR TITLE
Make list_chunkmanagers more resilient to broken entrypoints

### PR DESCRIPTION
**Pull request by hmaarrfk**
As I'm a developing my custom chunk manager, I'm often checking out between my development branch and production branch breaking the entrypoint.

This made xarray impossible to import unless I re-ran `pip install -e . -vv` which is somewhat tiring.

This should help xarray be more resilient in other software's bugs in case they install malformed entrypoints

Example:

```python
>>> from xarray.core.parallelcompat import list_chunkmanagers

>>> list_chunkmanagers()
<ipython-input-3-19326f4950bc>:1: UserWarning: Failed to load entrypoint MyChunkManager due to No module named 'my.array._chunkmanager'. Skipping.
  list_chunkmanagers()
{'dask': <xarray.core.daskmanager.DaskManager at 0x7f5b826231c0>}
```

Thank you for considering.

- [x] Closes #xxxx
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] New functions/methods are listed in `api.rst`

This is mostly a quality of life thing for developers, I don't see this as a user visible change.

### CODE REVIEW

**Comment by TomNicholas** [comment]
This looks great! Is it possible to test this error message somehow?

**Comment by hmaarrfk** [comment]
and rebased. please see suggested test.

File: xarray/tests/test_parallelcompat.py [hunk 1]
```diff
@@ -218,3 +220,13 @@ def test_raise_on_mixed_array_types(self, register_dummy_chunkmanager) -> None:
 
         with pytest.raises(TypeError, match="received multiple types"):
             get_chunked_array_type(*[dask_arr, dummy_arr])
+
+
+def test_bogus_entrypoint():
```
Comment (TomNicholas) [inline xarray/tests/test_parallelcompat.py:225]:
```suggestion
def test_bogus_entrypoint() -> None:
```
Let's see if this fixes the mypy error...

**Comment by TomNicholas** [comment]
Amazing! I've just pushed a commit to try to appease mypy.

**Comment by TomNicholas** [comment]
You can also put an entry in what's new if you want :)

**Comment by TomNicholas** [comment]
hmm - maybe adding ` -> None` to `utils.emit_user_level_warning` will work?

**Comment by hmaarrfk** [comment]
> You can also put an entry in what's new if you want :

let’s keep the scope of this fix small. I don’t think it warrants much of a message.

let’s keep the message for when we tackle the underlying challenge that brought me here ^_^

**Review by Illviljan — APPROVED** [review]

**Pull request merged**